lib/sedna/algorithms/multi_task_learning/multi_task_learning.py

```diff
@@ -0,0 +1,296 @@
+            if not callable(task_index['extractor']) and \
+                    isinstance(task_index['extractor'], str):
+                FileOps.download(task_index['extractor'], extractor_file)
+                self.extractor = joblib.load(extractor_file)
+            else:
+                self.extractor = task_index['extractor']
+            self.task_groups = task_index['task_groups']
+            self.models = [task.model for task in self.task_groups]
+        data, mappings = self.task_mining(samples=data)
+        samples, models = self.task_remodeling(samples=data, mappings=mappings)
+
+        callback = None
+        if post_process:
+            callback = ClassFactory.get_cls(ClassType.CALLBACK, post_process)()
+
+        tasks = []
+        for inx, df in enumerate(samples):
+            m = models[inx]
+            if not isinstance(m, Model):
+                continue
+            model_obj = set_backend(estimator=self.base_model)
+            evaluator = model_obj.load(m.model) if isinstance(m.model, str) else m.model
+            pred = evaluator.predict(df.x, kwargs=kwargs)
+            if callable(callback):
+                pred = callback(pred, df)
+            task = Task(entry=m.entry, samples=df)
+            task.result = pred
+            task.model = m
+            tasks.append(task)
+        res = self.inference_integrate(tasks)
+        return res, tasks
+
+    def evaluate(self, data: BaseDataSource,
+                 metrics=None,
+                 metrics_param=None,
+                 **kwargs):
+        from sklearn import metrics as sk_metrics
+
+        result, tasks = self.predict(data, kwargs=kwargs)
+        m_dict = {}
+        if metrics:
+            if callable(metrics):
+                m_name = getattr(metrics, '__name__', "mtl_eval")
+                m_dict = {
+                    m_name: metrics
+                }
+            elif isinstance(metrics, (set, list)):
+                for inx, m in enumerate(metrics):
+                    m_name = getattr(m, '__name__', f"mtl_eval_{inx}")
+                    if isinstance(m, str):
+                        m = getattr(sk_metrics, m)
+                    if not callable(m):
+                        continue
+                    m_dict[m_name] = m
+            elif isinstance(metrics, str):
+                m_dict = {
+                    metrics: getattr(sk_metrics, metrics, sk_metrics.log_loss)
+                }
+            elif isinstance(metrics, dict):
+                for k, v in metrics.items():
+                    if isinstance(v, str):
+                        v = getattr(sk_metrics, v)
+                    if not callable(v):
+                        continue
+                    m_dict[k] = v
+        if not len(m_dict):
+            m_dict = {
+                'precision_score': sk_metrics.precision_score
+            }
+            metrics_param = {"average": "micro"}
+
+        data.x['pred'] = result
+        data.x['y'] = data.y
```

Comment:
it maybe better understood that you use keys named "pred_y" and "real_y"

Reply:
fix